### PR TITLE
refactor(utils): migrate money utils and model out of extension

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 81aafe6 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9286567527), [Test report](https://leather-wallet.github.io/playwright-reports/refactor-use-money-model), [Storybook](https://refactor-use-money-model--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=refactor-use-money-model)

This pr migrates money utils and money model out of extension in the scope of https://github.com/leather-wallet/issues/issues/44

### CODE REVIEW

File: src/shared/models/currencies.model.ts [hunk 1]
```diff
@@ -3,5 +3,3 @@ import type { LiteralUnion } from 'leather-styles/types';
 export type CryptoCurrencies = LiteralUnion<'BTC' | 'STX', string>;
 
 export type FiatCurrencies = 'USD' | string;
-
-export type Currencies = CryptoCurrencies | FiatCurrencies;
```
Comment (pete-watters) [inline src/shared/models/currencies.model.ts:7]: Maybe the `currencies.model` needs to move to  `'@leather-wallet/models'` ?

**Review by pete-watters — APPROVED** [review]
🚀 Great stuff

File: src/shared/models/currencies.model.ts (hunk 1 above)
Comment (alter-eggo) [reply src/shared/models/currencies.model.ts:7]: it's there already

Comment (pete-watters) [reply src/shared/models/currencies.model.ts:7]: 👍 I meant maybe we can remove this `models/currencies.model.ts` file from the extension and take `CryptoCurrencies` + `FiatCurrencies` from the package?

Comment (alter-eggo) [reply src/shared/models/currencies.model.ts:7]: ah, sure, I just wanted to do this in sep pr

File: src/app/common/money/calculate-money.spec.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
+import { createMoney, createMoneyFromDecimal } from '@leather-wallet/utils';
```
Comment (fbwoolf) [inline src/app/common/money/calculate-money.spec.ts:1]: @kyranjamie wasn't there discussion about moving `createMoney` and `createMoneyFromDecimal` to the models pkg so it lives with the `Money` model?

Comment (alter-eggo) [reply src/app/common/money/calculate-money.spec.ts:1]: hm, I think like in `models` should live only `models` no?

Comment (fbwoolf) [reply src/app/common/money/calculate-money.spec.ts:1]: We had discussed (I think) having the functions that create the models live with the models, see `createMarketData` in `market.model.ts`.

Comment (alter-eggo) [reply src/app/common/money/calculate-money.spec.ts:1]: I see, makes sense

Comment (kyranjamie) [reply src/app/common/money/calculate-money.spec.ts:1]: I think it's okay for "model creation" fns to live with the models, but not logic more complicated than this

**Pull request merged**